### PR TITLE
use socrata connector, auth for ingest/library calls

### PR DESCRIPTION
**Pull request by fvankrieken**
Meant to close https://github.com/NYCPlanning/data-engineering/issues/1311

Seems that some dot datasets went private, meaning we can't simply download them without auth

2 commits just to make changes easier to see (1 just moves some logic, 2 makes needed changes)

passing run [here](https://github.com/NYCPlanning/data-engineering/actions/runs/12244696675)

### CODE REVIEW

File: dcpy/connectors/socrata/__init__.py [hunk 1]
```diff
@@ -0,0 +1,23 @@
+import os
+import requests
+from typing import Literal
+
+SOCRATA_USER = os.getenv("SOCRATA_USER")
+SOCRATA_PASSWORD = os.getenv("SOCRATA_PASSWORD")
+
+
+def _simple_auth():
```
Comment (sf-dcp) [inline dcpy/connectors/socrata/__init__.py]: can we move this and `_socrata_request()` to some utils ile?

**Comment by sf-dcp** [comment]
LGTM! The only rec is to move helper functions out of the init file

**Review by sf-dcp — APPROVED** [review]

File: dcpy/connectors/socrata/__init__.py (hunk 1 above)
Comment (fvankrieken) [reply dcpy/connectors/socrata/__init__.py]: Yup

File: dcpy/connectors/socrata/utils.py [hunk 2]
```diff
@@ -0,0 +1,25 @@
+import os
+import requests
+from typing import Literal
+
+SOCRATA_USER = os.getenv("SOCRATA_USER")
+SOCRATA_PASSWORD = os.getenv("SOCRATA_PASSWORD")
+
+
+def _simple_auth() -> tuple[str, str] | None:
```
[the remark this answers is not in the document]
Comment (fvankrieken) [reply dcpy/connectors/socrata/utils.py:9]: Going with no auth is fine for get requests for things like metadata or pulling datasets

[the remark this answers is not in the document]
Comment (fvankrieken) [reply dcpy/connectors/socrata/utils.py:9]: I think it makes sense to try to use auth when user/pass is defined, and let failures occur when a resource isn't allowed for a given user (or `None` auth).

Arguably then, maybe we only call simple_auth if those are defined. But I thought this was simple - get optional auth credentials if available.

**Review by alexrichey — APPROVED** [review]
cool - had one question, but this looks good

File: dcpy/connectors/socrata/utils.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (fvankrieken) [reply dcpy/connectors/socrata/utils.py:9]: Yup - just ran locally with no issue

**Pull request merged**